Fix misspellings (#6191)
Rdatatable/data.table: vignettes/datatable-importing.Rmd

## Care needed when providing and using options

> Common practice by R packages is to provide customization options set by `options(name=val)` and fetched using `getOption("name", default)`. Function arguments often specify a call to `getOption()` so that the user knows (from `?fun` or `args(fun)`) the name of the option controlling the default for that parameter; e.g. `fun(..., verbose=getOption("datatable.verbose", FALSE))`. All `data.table` options start with `datatable.` so as to not conflict with options in other packages. A user simply calls `options(datatable.verbose=TRUE)` to turn on verbosity. This affects all calls to `fun()` other the ones which have been provided `verbose=` explicitly; e.g. `fun(..., verbose=FALSE)`.

Reviewer: other than*

Reviewer: Also I'm not sure if ' ... been provided `verbose=` explicitly' makes sense (`verbose=` feels out of place)

Reviewer 2: Suggested wording:

> This affects all data.table function calls unless `verbose=FALSE` is provided explicitly.

Author: @MichaelChirico  Like that, keeping the example in the end?

```suggestion
Common practice by R packages is to provide customization options set by `options(name=val)` and fetched using `getOption("name", default)`. Function arguments often specify a call to `getOption()` so that the user knows (from `?fun` or `args(fun)`) the name of the option controlling the default for that parameter; e.g. `fun(..., verbose=getOption("datatable.verbose", FALSE))`. All `data.table` options start with `datatable.` so as to not conflict with options in other packages. A user simply calls `options(datatable.verbose=TRUE)` to turn on verbosity. This affects all data.table function calls unless `verbose=FALSE` is provided explicitly; e.g. `fun(..., verbose=FALSE)`.
```
